src/main/resources/alfresco/templates/webscripts/org/alfresco/repository/person/person.put.json.js

```diff
@@ -8,81 +8,85 @@ function main()
       status.setCode(status.STATUS_NOT_FOUND, "Person " + userName + " does not exist");
       return;
    }
-   
+
+   // MNT-21150 LDAP synced attributes can be changed using REST API
+   var qname = "{http://www.alfresco.org/model/content/1.0}";
+   var immutableProperties = people.getImmutableProperties(userName);
    // assign new values to the person's properties
-   if (!json.isNull("firstName"))
+
+   if ((!json.isNull("firstName")) && (!immutableProperties.hasOwnProperty((qname + "firstName").toString())))
```

Comment:
Not something you introduced, but I think we should just add a list with these properties and iterate through it, checking if not null and not immutable.